### PR TITLE
Check for EntityManagerInterface in Doctrine ORM data providers

### PR DESCRIPTION
**Pull request by ad3n**
Use EntityManagerInterface to consist with ODM

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | no
| License       | MIT
| Doc PR        | no

### CODE REVIEW

**Comment by dunglas** [comment]
I'm not sure it is worth it (and PHPStan doesn't like that)

**Comment by teohhanhui** [comment]
`phpstan` is not happy because you haven't changed the type at line 71.

I think this is a good change.

**Comment by ad3n** [comment]
@dunglas: i updated and test is passed

**Comment by teohhanhui** [comment]
Please target 2.4 branch instead.

**Comment by ad3n** [comment]
@teohhanhui : done

**Comment by teohhanhui** [comment]
It's not done correctly. I'll take care of it.

**Comment by ad3n** [comment]
@teohhanhui : ok thanks... sorry for misleading

File: src/Bridge/Doctrine/Orm/CollectionDataProvider.php [hunk 1]
```diff
@@ -45,7 +46,7 @@ public function __construct(ManagerRegistry $managerRegistry, iterable $collecti
 
     public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
     {
-        return null !== $this->managerRegistry->getManagerForClass($resourceClass);
+        return $this->managerRegistry->getManagerForClass($resourceClass) instanceof EntityManagerInterface;
```
Comment (teohhanhui) [inline src/Bridge/Doctrine/Orm/CollectionDataProvider.php]: @ad3n Actually we missed changing the phpdoc type below: https://github.com/api-platform/core/pull/3005/files#diff-fe622e94e81bf34426323e3ecc2e053eR59

Haha...

Comment (ad3n) [reply src/Bridge/Doctrine/Orm/CollectionDataProvider.php]: hahaha you are right, i just delete the line without adding the new one. Sorry for my mistake

Comment (ad3n) [reply src/Bridge/Doctrine/Orm/CollectionDataProvider.php]: I will create new PR

**Pull request merged**